tests/cli/main/test_check_file_output.py

```diff
@@ -130,22 +101,44 @@ def test_exists(
 
 @pytest.mark.parametrize("path", [pytest.param({"exists": True}, id="")], indirect=True)
 @pytest.mark.parametrize(
-    ("prompt", "exits"),
+    ("prompt", "exits", "log"),
     [
         pytest.param(
             {"ask": "y"},
             nullcontext(),
+            [
+                ("streamlink.cli", "info", "Writing output to\n/path/to/file"),
+                ("streamlink.cli", "debug", "Checking file output"),
+            ],
             id="yes",
         ),
         pytest.param(
             {"ask": "n"},
             pytest.raises(StreamlinkCLIError),
+            [
+                ("streamlink.cli", "info", "Writing output to\n/path/to/file"),
+                ("streamlink.cli", "debug", "Checking file output"),
+            ],
             id="no",
         ),
         pytest.param(
             {"ask": None},
             pytest.raises(StreamlinkCLIError),
-            id="error",
+            [
+                ("streamlink.cli", "info", "Writing output to\n/path/to/file"),
+                ("streamlink.cli", "debug", "Checking file output"),
+            ],
+            id="none",
+        ),
+        pytest.param(
+            {"ask": OSError()},
+            pytest.raises(StreamlinkCLIError),
+            [
+                ("streamlink.cli", "info", "Writing output to\n/path/to/file"),
+                ("streamlink.cli", "debug", "Checking file output"),
+                ("streamlink.cli", "error", "File file already exists, use --force to overwrite it."),
```

Comment:
Are we actually testing `--force` somewhere in terms of the file output to ensure it doesn't (or does) properly overwrite?

Reply:
See `test_exists`. I'll have a look later again, because some parametrization data was removed for this test because of the changed error handling, and the current setup is a bit weird now.